WordPress/Classes/Services/CommentService+Likes.swift

```diff
@@ -94,22 +94,19 @@ private extension CommentService {
             return
         }
 
-        let derivedContext = ContextManager.shared.newDerivedContext()
-
-        derivedContext.perform {
-
-            let likers = remoteLikeUsers.map { remoteUser in
-                LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: derivedContext)
-            }
+        Task {
+            await ContextManager.shared.save { derivedContext in
+                let likers = remoteLikeUsers.map { remoteUser in
```

Comment:
While it's not exceedingly likely that these objects will be mutated concurrently, it is possible here. It's (seemingly) stable under a queue-based concurrency model, but I'm not sure it's worth switching models until/unless we can guarantee thread safety. 

Interested to hear y'alls thoughts here 🙂

Reply:
My understanding is `ContextManager.shared.save` is called by a random queue managed by Swift's structured concurrency, whereas the closure here is called inside an embedded "NSManagedObjectContext queue" from `derivedContext.perform`? So, I don't think the way that this `remoteLikeUsers.map` statement is called is changed?

Reply:
Right, sorry, I probably could've made this a bit clearer – so this methods' first argument `remoteLikeUsers` (btw, accepting an optional array is also bad API design IMHO) is an array of reference types. Thus, they could be called in a manner similar to:

```swift
commentService.createNewUsers(from: likeUsers, commentID: 1, siteID: 1, purgeExisting: false) {
  // Do some stuff
} 

likeUsers.each { user in
  user.isCreating = true
}

``` 

It _doesn't_ work this way currently, but because these are value types we IMHO can't use them inside a `Task` this way because we have no guarantees about their state?